### ISSUE TITLE
[Icon] Make icons more accessible

### ISSUE THREAD

**Issue by billhimmelsbach** [cfpb/design-system-react#324]
Currently we use `alt` as the accessibility feature on `Icon`s that serve as images with the role `img`, but `alt` [isn't quite right for a `svg`](https://accessibilityinsights.io/info-examples/web/svg-img-alt/). So we have a couple options, and we can pick one of these depending on the use case: 

Instead of `alt`, the `svg` should have one of these:
  - a `<title>` + `<description>` as a child of the `<svg>` 
  - the `svg` could have an `aria-labelledby` and an optional `aria-describedby`
  - just an `aria-label`  for simple `svg`s 
  
I'd prefer to use title/description, but it would be tricker to manipulate/inject that into the raw html that this component relies on. I've allowed `aria-labelledby` and an optional `aria-describedby` in case people want them for more complicated `svg`s, but `aria-label` should serve most of our needs.

For some cases, the `svg` doesn't provide any additional information than the anchor link it's part of, in that case we want to use the new `isPresentational` to not announce the`svg` for users using screen readers.

## Current behavior
- alt doesn't provide enough accessibility for an svg

## Expected behavior
- add `aria-label`, `aria-labelledby`, `aria-describedby`, `isPresentational`
